gobblin-core/src/main/java/gobblin/writer/http/HttpWriterDecorator.java

```diff
@@ -0,0 +1,65 @@
+/**
+ *
+ */
+package gobblin.writer.http;
+
+import java.io.IOException;
+
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.ListenableFuture;
+
+/**
+ * Common parent for {@link AbstractHttpWriter} decorators. Delegates extension methods to another
+ * implementation and simplifies the overriding of only selected methods
+ */
+public abstract class HttpWriterDecorator<D> implements HttpWriterDecoration<D> {
```

Comment:
Yeah. Although it does not have any abstract methods, I don't see a point of instantiating this class.